src/main/java/hudson/remoting/Channel.java

```diff
@@ -1820,6 +1861,55 @@ public static void dumpDiagnosticsForAll(@Nonnull PrintWriter w) {
         }
     }
 
+    /**
+     * Notification that {@link Command#readFrom} has succeeded.
+     * @param cmd the resulting command
+     * @param blockSize the serialized size of the command
+     * @see CommandListener
+     */
+    void notifyRead(Command cmd, long blockSize) {
```

Comment:
🐛  for this and all following handers. It is a de-facto extension point offered in API. An improper implementation of the extension point may bring down the channel  and cause other undefined behavior. We have already had much fun inside Jenkins due to such kind of listeners (including DoS when bogus ComputerListeners were blocking Jenkins startups).

It may be feasible to NOT handle the exceptions due to performance reasons, but in such case it should be documented in `onSomething()` Javadocs at least .

Reply:
Agreed, can catch unchecked exceptions here.

Reply:
Maybe it makes sense to measure the performance impact after that. https://github.com/jenkinsci/remoting/tree/master/src/assembly/agent-load-test is probably enough to make some estimations if you run it with a profiler